uaclient/timer/tests/test_update_contract_info.py

```diff
@@ -84,3 +87,61 @@ def test_contract_failure(
             "Failed to check for change in machine contract."
             " Reason: Error checking contract info\n"
         ) in caplog_text()
+
+
+class TestValidateReleaseSeries:
+    @pytest.mark.parametrize(
+        "entitlements, current_series",
+        [
+            (
+                {
+                    "support": {
+                        "entitlement": {"affordances": {"onlySeries": None}}
+                    }
+                },
+                "jammy",
+            ),
+            (
+                {
+                    "support": {
+                        "entitlement": {"affordances": {"onlySeries": "jammy"}}
+                    }
+                },
+                "jammy",
+            ),
+            (
+                {
+                    "support": {
+                        "entitlement": {
+                            "affordances": {"onlySeries": "bionic"}
+                        }
+                    }
+                },
+                "jammy",
+            ),
+        ],
+    )
+    @mock.patch(
+        "uaclient.system.get_release_info",
+        return_value=mock.MagicMock(series="jammy"),
+    )
+    @mock.patch(M_PATH + "detach")
+    def test_validate_release_series(
+        self,
+        m_detach,
+        m_get_release_info,
+        FakeConfig,
+        fake_machine_token_file,
+        entitlements,
+        current_series,
+    ):
+        allowed_series = entitlements["support"]["entitlement"]["affordances"][
+            "onlySeries"
+        ]
+        fake_machine_token_file._entitlements = entitlements
+        validate_release_series(cfg=FakeConfig())
+        if allowed_series:
+            if current_series != allowed_series:
+                assert m_detach.call_count == 1
+        else:
+            assert m_detach.call_count == 0
```

Comment:
I tried to make it a little simpler, only change if you agree
```suggestion
class TestValidateReleaseSeries:
    @pytest.mark.parametrize("allowed_series", (None, "bionic", "jammy"))
    @mock.patch(
        "uaclient.system.get_release_info",
        return_value=mock.MagicMock(series="jammy"),
    )
    @mock.patch(M_PATH + "detach")
    def test_validate_release_series(
        self,
        m_detach,
        m_get_release_info,
        allowed_series,
        FakeConfig,
        fake_machine_token_file,
    ):
        fake_machine_token_file._entitlements = {
            "support": {
                "entitlement": {"affordances": {"onlySeries": allowed_series}}
            }
        }

        validate_release_series(cfg=FakeConfig())
        if allowed_series:
            if allowed_series != "jammy":
                assert m_detach.call_count == 1
        else:
            assert m_detach.call_count == 0
```

Reply:
wow huge suggestions look horrible in the page

Reply:
Looks much better with the change